exercises/bank-account/src/test/java/BankAccountTest.java

```diff
@@ -0,0 +1,201 @@
+
+        bankAccount.withdraw(5);
+    }
+
+    @Ignore
+    @Test
+    public void cannotWithdrawMoreMoneyThanYouHave() throws BankAccountActionInvalidException {
+        bankAccount.open();
+        bankAccount.deposit(6);
+
+        expectedException.expect(BankAccountActionInvalidException.class);
+        expectedException.expectMessage("Cannot withdraw more money than is currently in the account");
+
+        bankAccount.withdraw(7);
+    }
+
+    @Ignore
+    @Test
+    public void cannotDepositNegativeAmount() throws BankAccountActionInvalidException {
+        bankAccount.open();
+
+        expectedException.expect(BankAccountActionInvalidException.class);
+        expectedException.expectMessage("Cannot deposit or withdraw negative amount");
+
+        bankAccount.deposit(-1);
+    }
+
+    @Ignore
+    @Test
+    public void cannotWithdrawNegativeAmount() throws BankAccountActionInvalidException {
+        bankAccount.open();
+        bankAccount.deposit(105);
+
+        expectedException.expect(BankAccountActionInvalidException.class);
+        expectedException.expectMessage("Cannot deposit or withdraw negative amount");
+
+        bankAccount.withdraw(-5);
+    }
+
+    @Ignore
+    @Test
+    public void cannotGetBalanceOfClosedAccount() throws BankAccountActionInvalidException {
+        bankAccount.open();
+        bankAccount.deposit(10);
+        bankAccount.close();
+
+        expectedException.expect(BankAccountActionInvalidException.class);
+        expectedException.expectMessage("Account closed");
+
+        bankAccount.getBalance();
+    }
+
+    @Ignore
+    @Test
+    public void cannotDepositMoneyIntoClosedAccount() throws BankAccountActionInvalidException {
+        bankAccount.open();
+        bankAccount.close();
+
+        expectedException.expect(BankAccountActionInvalidException.class);
+        expectedException.expectMessage("Account closed");
+
+        bankAccount.deposit(5);
+    }
+
+    @Ignore
+    @Test
+    public void cannotWithdrawMoneyFromClosedAccount() throws BankAccountActionInvalidException {
+        bankAccount.open();
+        bankAccount.deposit(20);
+        bankAccount.close();
+
+        expectedException.expect(BankAccountActionInvalidException.class);
+        expectedException.expectMessage("Account closed");
+
+        bankAccount.withdraw(5);
+    }
+
+    @Ignore
+    @Test
+    public void bankAccountIsClosedBeforeItIsOpened() throws BankAccountActionInvalidException {
+        expectedException.expect(BankAccountActionInvalidException.class);
+        expectedException.expectMessage("Account closed");
+
+        bankAccount.getBalance();
+    }
+
+    @Ignore
+    @Test
+    public void canAdjustBalanceConcurrently() throws BankAccountActionInvalidException, InterruptedException {
+        bankAccount.open();
+        bankAccount.deposit(1000);
+
+        for (int i = 0; i < 10; i++) {
+            adjustBalanceConcurrently();
```

Comment:
It failed every time I ran it (I think I ran it about 10 times in a row) with a non-threadsafe implementation. Is there any way we can be absolutely guarantee a concurrent modification attempt? I would expect the test to fail every time if their implementation isn't thread safe, but I can't be 100% certain. I could put in the hint that they should run the test a couple of times just to be safe?